tests/rptest/services/compatibility/example_runner.py

```diff
@@ -25,17 +27,28 @@ def __init__(self, context, example, timeout_sec=10):
 
         self._pid = None
 
+        self._stopping = threading.Event()
+
     def _worker(self, idx, node):
-        start_time = time.time()
+        self._stopping.clear()
 
         # Some examples require the hostname of the node
         self._example.set_node_name(node.name)
 
         # Run the example until the condition is met or timeout occurs
         cmd = "echo $$ ; " + self._example.cmd()
         output_iter = node.account.ssh_capture(cmd)
-        while not self._example.condition_met(
-        ) and time.time() < start_time + self._timeout:
+
+        start_time = time.time()
+
+        while True:
+
+            # Terminate loop on timeout or stop_node
```

Comment:
It seems like you could still have a self.condition_met condition for dropping out (as well as stopping +timeout).  Otherwise we'll keep looping here until stop() is called -- or is that the ideal, to keep capturing output?